src/Shared/Targets/Wrappers/AspNetBufferingTargetWrapper.cs

```diff
@@ -223,14 +223,14 @@ private Internal.LogEventInfoBuffer GetOrCreateRequestBuffer(HttpContextBase con
 
             lock (bufferDictionary)
             {
-                if (!bufferDictionary.ContainsKey(this))
+                Internal.LogEventInfoBuffer buffer;
+                if (!bufferDictionary.TryGetValue(this, out buffer))
                 {
                     bufferDictionary.Add(this,
                         new Internal.LogEventInfoBuffer(BufferSize, GrowBufferAsNeeded, BufferGrowLimit));
                 }
+                return bufferDictionary[this];
```

Comment:
maybe do this instead (single lookup):
```c#
                if (!bufferDictionary.TryGetValue(this, out buffer))
                {
                    buffer = new Internal.LogEventInfoBuffer(BufferSize, GrowBufferAsNeeded, BufferGrowLimit)
                    bufferDictionary.Add(this, buffer);
                }
                return buffer;
```

Reply:
Completed

Reply:
Please re-add the `lock (bufferDictionary)` around this logic

Reply:
Sorry about that.

Reply:
No worries. Now just need to find some volunteers, that find it useful, and wants to test it.